plugins/languages/treesitter/treesitter.nix

```diff
@@ -12,6 +12,103 @@ helpers.neovim-plugin.mkNeovimPlugin config {
   luaName = "nvim-treesitter.configs";
   defaultPackage = pkgs.vimPlugins.nvim-treesitter;
 
+  description = ''
+    Provides an interface to [tree-sitter]
+
+    ### Installing Your Own Grammars with Nixvim
+
+    The `nvim-treesitter` derivation generates the grammar packages based on the
+    default configs of nvim-treesitter but you can always install your own
+    grammars.
```

Comment:
I think this is confusingly worded. `pkgs.vimPlugins.nvim-treesitter` is a derivation (all packages are), but the `plugins.treesitter` module is _not_ a derivation.

I'm not sure it's strictly accurate to say the package derivation generates the grammar derivations, since that is done in this module by providing an overridden package using the `withPlugins` function.

Not sure the best way to word it though.

Reply:
How about:
```
If the grammar you want to install isn't provided through `nixpkgs`, you could install it like so:
```

Reply:
I think it's nice to keep an implication that most users don't need to do anything, and that configuring `plugins.treesitter.nixGrammars` and/or `plugins.treesitter.grammarPackages` should cover 80% of usage when users _do_ need to do something.

> If the grammar you want to install isn't available through `nixpkgs`, you could install it like so:

But something like that could definitely fit into the text!

Reply:
> I think it's nice to keep an implication that most users don't need to do anything

If that's the case, these docs might be a better fit in some other document since it is sort of a niche case. Having it front and centre of such a ubiquitous plugin might scare people off 👀

Reply:
I think it's fine to have it here. Let's not let perfect be the enemy of good!

Reply:
I've reworded it. Let me know if that works for you.